### PR TITLE
Feature rabbit publisher builder

### PR DESCRIPTION
**Pull request by ybs007**

### CODE REVIEW

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+namespace Blizzard.IO.RabbitMQ.Builders
+{
+    public class NetqRabbitPublisherBuilder<TData> : BaseNetqRabbitBuilder
+    {
+        private RabbitExchange _destinationExchange = null;
+        private ISerializer<TData> _serializer = new JsonSerializer<TData>();
+        private string _routingKey = "/";
+        private bool _isAbstract;
+        private IConverter<RabbitMessageProperties, MessageProperties> _converter;
+        private ILoggerFactory _loggerFactory;
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: Why do you have your own logger factory where the base has another logger factory?

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            Product = product;
+            Platform = platform;
+            PublisherConfirms = publisherConfirms;
+            RequestHeartbeat = requestHeartbeat;
+            VirtualHost = virtualHost;
+            PersistentMessages = persistentMessages;
+            return this;
+        }
+
+        public NetqRabbitPublisherBuilder<TData> AddHost(string host)
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: Why do you have a separate host function?

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            return this;
+        }
+
+        public NetqRabbitPublisherBuilder<TData> AddRoutingKey(string routingKey)
+        {
+            _routingKey = routingKey;
+            return this;
+        }
+
+        public NetqRabbitPublisherBuilder<TData> AddLoggerFactory(ILoggerFactory loggerFactory)
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: It seems redundant you have one factory per application

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        public NetqRabbitPublisherBuilder<TData> AddConverter(
+            IConverter<RabbitMessageProperties, MessageProperties> converter)
+        {
+            _converter = converter;
+            return this;
+        }
+
+        public NetqRabbitPublisher<TData> BuildPublisher()
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: You can just call it build

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs (hunk 1 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: f

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs (hunk 2 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: f

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs (hunk 3 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: f

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs (hunk 4 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: f

File: Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+using Microsoft.Extensions.Logging;
+
+namespace Blizzard.IO.RabbitMQ.Builders
+{
+    public class NetqRabbitPublisherBuilder<TData> : BaseNetqRabbitBuilder
+    {
+        private RabbitExchange _destinationExchange = null;
+        private ISerializer<TData> _serializer = new JsonSerializer<TData>();
+        private string _routingKey = "/";
+        private bool _isAbstract;
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: Initialize this field

Comment (ybs007) [reply Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: The default of bool is false, so there is no need to initialize it

Comment (alloncm) [reply Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: I know but make it more readable

Comment (ybs007) [reply Blizzard.IO.RabbitMQ.Builders/NetqRabbitPublisherBuilder.cs]: f

**Pull request merged**